src/hooks/useMouseIntent.js

```diff
@@ -0,0 +1,37 @@
+/* adapted from: https://github.com/github/github/blob/a959c0d15c29b98c49b881f520c5947fe24eecb9/app/assets/modules/github/behaviors/button-outline.ts */
+import {useEffect} from 'react'
+
+function useMouseIntent() {
+  useEffect(() => {
+    let lastActiveElement = null
+    let currentInputIsMouse = false
+
+    function setClass() {
+      lastActiveElement = document.activeElement
+      if (document.body) {
+        document.body.classList.toggle('intent-mouse', currentInputIsMouse)
+      }
+    }
+    // Use mousedown event to make sure outline is remove for holding and dragging
+    document.addEventListener(
+      'mousedown',
+      function() {
+        currentInputIsMouse = true
+        if (lastActiveElement === document.activeElement) setClass()
+      },
+      {capture: true}
+    )
+
+    document.addEventListener(
+      'keydown',
+      function() {
+        currentInputIsMouse = false
+      },
+      {capture: true}
+    )
+
+    document.addEventListener('focusin', setClass, {capture: true})
+  })
```

Comment:
🤔 I don't know for sure here, so I'm kinda shooting from the hip a bit with these thoughts, but here they are off the top of my head:

1. Should the dependencies list here be set to an empty array to ensure this is only ever run once?
2. Should this function return a teardown function to remove the event listeners if the rendering component is removed?
3. Should we do anything to prevent this hook being used more than once on a page? 
4. Is a hook the right thing for this? Could it just be a function that is called by the application once outside of the react component tree?

Reply:
I think it makes sense to keep this as a hook so we can return a cleanup function that removes all the event listeners from the `document` when the component is unmounted. `BaseStyles` seems like a fitting place to call this hook.